### DISCUSSION TITLE
[idea] server blacklist

### DISCUSSION THREAD

**Discussion by spcqike** [alexjustesen/speedtest-tracker#17] [Ideas]
Hey 

thanks for your effort. 
as you mentioned in the original repo, I would like to recommend one feature wish:
Create a blacklist of servers. 

https://github.com/henrywhitaker3/Speedtest-Tracker/issues/806

Best regards

**Comment by alexjustesen** [comment]
@spcqike looks like @bdowden is already working on that. Spoke to him earlier and we'll be getting this into the app for sure.

PR should you want to follow along: https://github.com/alexjustesen/speedtest-tracker/pull/16

**Reply by MeadowValley** [reply]
What is the current state? PR was closed and I don't see a follow-up.
Edit: Nevermind, I found https://github.com/alexjustesen/speedtest-tracker/issues/1877

**Comment by bdowden** [comment]
Looks like `--exclude` isn't part of the official CLI now. The idea I'm going with is to filter out servers that are closest to the user (via the `-L` flag), then selecting the first server in that result set.

**Comment by alexjustesen** [comment]
Listing preferred speedtest servers is available, so closing this one.

**Reply by gabeklavans** [reply]
In my case, I'd still like a blocklist functionality. There's just one specific server that gets automatically selected sometimes that tanks my graphs, likely due to some problem between me and their server. My workaround is to add all the other servers that do work fine to my allowlist, but as a result, around 10% of the time, I get this error:
```
{"type":"log","timestamp":"2024-01-14T02:00:03Z","message":"Configuration - No servers defined (NoServersException)","level":"error"}
```
which messes up my history even more than before, with 0s.

I'm assuming it's because my allowlist isn't exhaustive and all the servers on it are busy at that moment. Unless I'm missing something, I think a block list feature would fix this problem.

**Comment by hcgatewood** [comment]
Something like this would actually be super helpful @alexjustesen -- I have the same issue where a single server is always causing notifications, but it's not annoying enough that I'll go through the effort of curating my list of servers (allow-list style). Instead, it would be easier to just click an offending server and block it (block-list style), directly from the UI.

Thoughts on potentially re-opening this issue for support? Happy to tip/sponsor if that would be helpful, thanks for a great application!

**Reply by alexjustesen** [reply]
#1831 provided a point where I can hook into the speedtest process and extend how servers are chosen. So the short answer is, this is now possible.

I created #1877 so you can track the research and progress on this feature.
